infra/Pos/Communication/Types/Protocol.hs

```diff
@@ -77,6 +81,22 @@ data Conversation m t where
         => (N.ConversationActions snd rcv m -> m t)
         -> Conversation m t
 
+newtype PeerId = PeerId ByteString
```

Comment:
What is `PeerId`? Is it needed? Currently it's not used anywhere. To be more precise, it's used in `TransactionProvenance` which is used in `MemPoolModifyReason`, but it's not used at all.